### DISCUSSION TITLE
[BUG]ReplyKeyboardMarkup doesn't work on desktop and Web app in ChatType.SUPERGROUP

### DISCUSSION THREAD

**Discussion by r-karasev** [python-telegram-bot/python-telegram-bot#4426] [Q&A] [answered]
### Steps to Reproduce

This code works fine on Android app, but doesn't work (keyboard doesn't appear) on desktop and web app (I tried on Linux and MacOS)

```
    async def start(self, update: Update, context: ContextTypes.DEFAULT_TYPE):
        custom_keyboard = [['top-left', 'top-right'], ['bottom-left', 'bottom-right']]
        reply_markup = ReplyKeyboardMarkup(custom_keyboard,
                                           resize_keyboard=True)

        await update.message.reply_text(text="Custom Keyboard Test",
                                        reply_markup=reply_markup)
```
Keyboard appears only in direct chat with the bot, but not in group chat.
Any ideas?

### Expected behaviour

User who wrote '/start' sees a keyboard

### Actual behaviour

User sees only message 'Custom Keyboard Test' with no keyboard

### Operating System

Linux, MacOS

### Version of Python, python-telegram-bot & dependencies

```shell
python-telegram-bot 21.4
Bot API 7.7
```

**Comment by harshil21** [comment]
Hello. This is not an issue with the PTB library. Please check that the keyboard icon is visible on the desktop app. Try to use the parameter https://docs.python-telegram-bot.org/en/doc-fixes/telegram.replykeyboardmarkup.html#telegram.ReplyKeyboardMarkup.params.is_persistent as well.

**Comment by r-karasev** [comment]
`is_persistent` doesn't help(
This is how it looks like
![111](https://github.com/user-attachments/assets/58bc78d4-0127-48ca-8b29-0e55040c8667)

And this in direct chat with the bot
![image](https://github.com/user-attachments/assets/034422c9-7096-4bae-afa2-abfb45b9057d)

**Reply by harshil21** [reply]
And does the parameter https://docs.python-telegram-bot.org/en/doc-fixes/telegram.replykeyboardmarkup.html#telegram.ReplyKeyboardMarkup.params.selective help?

**Reply by r-karasev** [reply]
No, this is what I tried
```
        reply_markup = ReplyKeyboardMarkup(custom_keyboard,
                                           selective=True,
                                           resize_keyboard=True,
                                           is_persistent=True)
```
Also I tried to add/remove `resize_keyboard` param.

**Comment by r-karasev** [comment]
This example doesn't work too 
https://github.com/python-telegram-bot/python-telegram-bot/blob/master/examples/conversationbot.py
It only works in direct chat with the bot, not in group chat.

**Comment by r-karasev** [comment]
Hi, I tried raw HTTPs request and it doesn't work
```
https://api.telegram.org/bot<TOKEN>/sendMessage?chat_id=<ID>&message_thread_id=<ID>&text=test&reply_markup=%7B%22keyboard%22%3A+%5B%5B%22Button%22%5D%5D%7D
```
but it works if I send a message direct to myself. So looks like there is a problem on Telegram side. 
Sorry to bother you!

**Comment by r-karasev** [comment, marked as answer]
This is Telegram bug https://bugs.telegram.org/c/27693
